extension/src/controllers/video-data-sync-controller.ts

```diff
@@ -158,30 +165,16 @@ export default class VideoDataSyncController {
         }
 
         const subtitleTrackChoices = this._syncedData?.subtitles ?? [];
-        const selectedSub =
-            this.lastLanguageSynced === ''
-                ? {
-                      language: '',
-                      url: '-',
-                      label: i18n.t('extension.videoDataSync.emptySubtitleTrack'),
-                      extension: 'srt',
-                  }
-                : subtitleTrackChoices.find((subtitle) => subtitle.language === this.lastLanguageSynced);
+        const subs = this._matchLastSyncedWithAvailableTracks();
+        const selectedSub: VideoDataSubtitleTrack[] = subs.autoSelectedTracks;
+        const defaultCheckboxState: boolean = subs.completeMatch;
 
         if (selectedSub !== undefined && !userRequested && !this._syncedData?.error) {
```

Comment:
With the current logic we will never get into the `else` block. The `else` block is used to show the track selector in case the user's preferred tracks aren't found. Right now, we are auto-loading tracks even when there are no matching tracks found.